tests/nativeapp/test_manager_teardown.py

```diff
@@ -0,0 +1,334 @@
+    side_effects, expected = mock_execute_helper(
+        [
+            # teardown app: app does not exist + is skipped
+            (
+                mock_cursor([{"CURRENT_ROLE()": "old_role"}], []),
+                mock.call("select current_role()", cursor_class=DictCursor),
+            ),
+            (None, mock.call("use role app_role")),
+            (
+                mock_cursor([], []),
+                mock.call("show applications like 'MYAPP'", cursor_class=DictCursor),
+            ),
+            (None, mock.call("use role old_role")),
+            # teardown package: package exists + is dropped
+            (
+                mock_cursor([{"CURRENT_ROLE()": "old_role"}], []),
+                mock.call("select current_role()", cursor_class=DictCursor),
+            ),
+            (None, mock.call("use role package_role")),
+            (
+                mock_cursor(
+                    [
+                        {
+                            "name": "APP_PKG",
+                            "comment": SPECIAL_COMMENT,
+                            "owner": "PACKAGE_ROLE",
+                        }
+                    ],
+                    [],
+                ),
+                mock.call(
+                    "show application packages like 'APP_PKG'",
+                    cursor_class=DictCursor,
+                ),
+            ),
+            (None, mock.call("drop application package app_pkg")),
+            (None, mock.call("use role old_role")),
+        ]
+    )
+    mock_execute.side_effect = side_effects
+
+    current_working_directory = os.getcwd()
+    create_named_file(
+        file_name="snowflake.yml",
+        dir=current_working_directory,
+        contents=[mock_snowflake_yml_file],
+    )
+
+    native_app_manager = NativeAppManager()
+    native_app_manager.teardown()
+    assert mock_execute.mock_calls == expected
+
+
+@mock.patch(NATIVEAPP_MANAGER_EXECUTE)
+def test_teardown_could_not_drop_app(mock_execute, temp_dir, mock_cursor):
+    side_effects, expected = mock_execute_helper(
+        [
+            # teardown app: app has wrong comment; drop throws an error
+            (
+                mock_cursor([{"CURRENT_ROLE()": "old_role"}], []),
+                mock.call("select current_role()", cursor_class=DictCursor),
+            ),
+            (None, mock.call("use role app_role")),
+            (
+                mock_cursor(
+                    [
+                        {
+                            "name": "MYAPP",
+                            "comment": "some other random comment",
+                            "version": "UNVERSIONED",
+                            "owner": "APP_ROLE",
+                        }
+                    ],
+                    [],
+                ),
+                mock.call("show applications like 'MYAPP'", cursor_class=DictCursor),
+            ),
+            (None, mock.call("use role old_role")),
+            # teardown package: never happens
```

Comment:
How about the usecase where drop happens but fails? Do we have an existing test for that?